server/main.go

```diff
@@ -62,20 +65,21 @@ func main() {
 		log.Fatalf("Failed to create meter: %v", err)
 	}
 
-	// If the user has set a database user or password, init the database
-	if os.Getenv("DB_USER") != "" || os.Getenv("DB_PASSWORD") != "" {
-		config.InitDB()
-	}
+	var proxyPort int
+	var resourceRepository interfaces.IRepository
+	var oauthService *oauthSvc.Service
+
+	if *InMemoryDb {
+		proxyPort = *port
```

Comment:
instead of put the pointer of variable port value to the proxyPort just set the SERVER_PORT env variable from the port value then at line 130 you just need to run the http server using `SERVER_PORT` env variable without pass it as an argument to the server function

Reply:
by doing this you also can delete line 101-105

Reply:
@ternakkode, are you suggesting this, 
`os.Setenv("SERVER_PORT", strconv.Itoa(*port))`?
instead of:
```
proxyPort = *port
os.Setenv("SERVER_PORT", strconv.Itoa(proxyPort))
```
If this works I would agree

Reply:
Agree, this makes sense.